web/src/beta/lib/reearth-ui/components/Selector/index.tsx

```diff
@@ -0,0 +1,229 @@
+import { FC, MouseEvent, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import { styled, useTheme } from "@reearth/services/theme";
+
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+import { Popup } from "../Popup";
+import { Typography } from "../Typography";
+
+export type SelectorProps = {
+  multiple?: boolean;
+  value?: string | string[];
+  options: { value: string; label?: string }[];
+  disabled?: boolean;
+  placeholder?: string;
+  onChange?: (value: string | string[]) => void;
+};
+
+export const Selector: FC<SelectorProps> = ({
+  multiple,
+  value,
+  options,
+  placeholder = "Please select",
+  disabled,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const selectorRef = useRef<HTMLDivElement>(null);
+  const [selectedValue, setSelectedValue] = useState<string | string[] | undefined>(
+    value ?? (multiple ? [] : undefined),
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectorWidth, setSelectorWidth] = useState<number>();
+
+  const optionValues = useMemo(() => options, [options]);
+
+  useEffect(() => {
+    setSelectedValue(value ?? (multiple ? [] : undefined));
+  }, [value, multiple]);
+
+  useEffect(() => {
+    setSelectorWidth(selectorRef.current?.clientWidth);
+  }, [selectorRef]);
```

Comment:
Maybe this is not enough. We may need a resizeObserver here, as you can imagin in reearth UI the side panel is resizeable, and the width could be changed anytime by user, we need to update this width once the selector width changes.